frontend/packages/dev-console/src/utils/pipeline-filter-reducer.ts

```diff
@@ -1,26 +1,13 @@
 import * as _ from 'lodash';
 
-export const pipelineFilterReducer = (pipeline): string => {
-  if (
-    !pipeline ||
-    !pipeline.latestRun ||
-    !pipeline.latestRun.status ||
-    !pipeline.latestRun.status.succeededCondition
-  ) {
-    return '-';
-  }
-  return pipeline.latestRun.status.succeededCondition;
-};
-
 export const pipelineRunStatus = (pipelineRun): string => {
-  if (
-    !pipelineRun ||
-    !pipelineRun.status ||
-    !pipelineRun.status.conditions ||
-    pipelineRun.status.conditions.length === 0
-  ) {
-    return null;
+  const conditions = _.get(pipelineRun, ['status', 'conditions'], []);
+  const isCancelled = conditions.find((c) => c.reason === 'PipelineRunCancelled');
+  if (isCancelled) {
+    return 'Cancelled';
   }
+  if (conditions.length === 0) return null;
+
   const condition = pipelineRun.status.conditions.find((c) => c.type === 'Succeeded');
```

Comment:
Reuse `conditions`
```suggestion
   const condition = conditions.find((c) => c.type === 'Succeeded');
```

Reply:
Ahh, missed that. Thanks :smile: Updated